core/src/components/input/input.scss

```diff
@@ -217,12 +218,13 @@
  * However, the clear button always disappears after
  * being activated, so we never get to that state.
  */
-.input-clear-icon:focus {
+.input-clear-icon:focus,
+.input-password-toggle:focus {
   opacity: 0.5;
 }
 
-:host(.has-value) .input-clear-icon {
-  visibility: visible;
+:host(:not(.has-value)) .input-clear-icon {
+  visibility: hidden;
```

Comment:
Why was this change made?

Reply:
Because otherwise, the clear icon would not disappear when the input is erased.

Reply:
The clear icon currently disappears when the input is erased on the latest stable version of Ionic: https://ionicframework.com/docs/api/input#clear-options

If that's not the case in this branch, then there may be other code impacting that.